enable fillMode to support an nth-column option

chbonser opened the pull request:
https://github.com/Addepar/ember-table/issues/634

![screen shot 2018-11-26 at 8 45 35 am](https://user-images.githubusercontent.com/1022582/49021304-f46c7f00-f157-11e8-99c2-51bd3393a90c.png)

cyril-sf commented:
@chbonser sorry for the delay.
It seems like a good option. Could you update your PR

chbonser commented:
@cyril-sf Yes. I can make those changes. It will be a few weeks though...

cyril-sf commented:
@chbonser no problem and thanks for the help!

billdami commented:
would love to see this merged in. I could possibly help out getting this across the finish line if you still don't have the bandwidth to work on it @chbonser ?

chbonser commented:
Sorry for the delay. I will make time for it next week.

On Tue, Jul 9, 2019 at 2:02 PM Bill Dami <notifications@github.com> wrote:

> would love to see this merged in. I could possibly help out getting this
> across the finish line if you still don't have the bandwidth to work on it
> @chbonser <https://github.com/chbonser> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Addepar/ember-table/pull/635?email_source=notifications&email_token=AAHZU5TH66C6JUKSKDJPCHLP6THEFA5CNFSM4GGMWWS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZRBQTI#issuecomment-509745229>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHZU5UUJXFSLVHOMHHWJODP6THEFANCNFSM4GGMWWSQ>
> .
>

chbonser commented:
@cyril-sf I've rebased and updated the fill mode implementation to reuse the resize implementation across all single column fill modes. The behavior of the preexisting modes remains unchanged so this is a non-breaking addition.

bantic commented on addon/components/ember-thead/component.js:
```diff
@@ -118,12 +118,19 @@ export default Component.extend({
   fillMode: defaultTo(FILL_MODE.EQUAL_COLUMN),
 
+  /**
+    A configuration that controls how which column shinks (or extends) when `fillMode` is
```
```suggestion
    A configuration that controls which column shrinks (or extends) when `fillMode` is
```

bantic commented on addon/components/ember-thead/component.js:
```diff
@@ -118,12 +118,19 @@ export default Component.extend({
+  /**
+    A configuration that controls how which column shinks (or extends) when `fillMode` is
+    'nth-column'. This is zero indexed.
+  */
```
For the autogenerated API docs can you add:

```
@argument fillColumnIndex
@type number?
```

cyril-sf commented:
@chbonser no need to apologize, we appreciate the extra effort!

bantic commented:
released in `v2.1.0`, thanks!
